### PR TITLE
HDF archive for multi array.

### PR DESCRIPTION
**Pull request by fdmalone**
@mmorale3

### CODE REVIEW

File: src/io/hdf_archive.h [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include <io/hdf_stl.h>
 #include <io/hdf_hyperslab.h>
 #include <io/hdf_double_hyperslab.h>
+#include <io/hdf_multi.h>
```
Comment (ye-luo) [inline src/io/hdf_archive.h]: Please move this inside HAVE_LIBBOOST

Comment (ye-luo) [reply src/io/hdf_archive.h]: Please also clean up hdf_ma.h. I guess it is legacy now.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -120,6 +120,10 @@ if (HAVE_MPI)
   INCLUDE_DIRECTORIES("${qmcpack_SOURCE_DIR}/external_codes/mpi_wrapper")
 endif()
 
+if (HAVE_LIBBOOST)
```
Comment (ye-luo) [inline src/CMakeLists.txt]: Change my mind a bit. MA is not part of boost yet. It must have its own macro if needed.
At the moment, MA is shipped within the code and used by default. We don't need to use a macro.
@markdewing what do you think?

File: src/io/hdf_archive.h (hunk 1 above)
Comment (ye-luo) [reply src/io/hdf_archive.h]: Changed my mind on put it inside HAVE_LIBBOOST macro. See above.

**Comment by mmorale3** [comment]
MA is actually boost's multi_array. multi is Alfredo' library, so
technically hdf_ma could be guarded.
In any case, boost is a requirement, no?

Miguel

**Comment by ye-luo** [comment]
I was so confused by boost multi_array and Alfredo's boost-multi.
Is Alredo's multi now included as external_codes/boost_multi?
Since it is not part of boost, guard it with HAVE_LIBBOOST in src/CMakeLists.txt is inappropriate.

Are both multi_array libraries simultaneously used in the AFQMC code?

Boost is a requirement at the moment.

**Comment by mmorale3** [comment]
Alfredo’s multi is now included with QMCPACK, but I haven’t finished replacing Boost’s multi_array with it yet. This is coming very soon, probably within the next 2-3 weeks, but it is not done yet. So for a bit longer, both libraries are used simultaneously.

**Pull request merged**